### PR TITLE
feat: add support `ollama` api

### PR DESCRIPTION
**Pull request by davidberenstein1957**
In this PR I've included basic support for [ollama](https://ollama.ai/)

Some design choices:

- [x] use builtin `urllib` library. we might replace this with `requests` to also allow for passing a `client`
- [x] retry 5 times with n+1 backoff in case of 500 (server errors) as fixed config for some robustness but to avoid spending a lot of time on this
- [x] default to `http://localhost:11434` but overwrite with `OLLAMA_HOST` env var as in Ollama docs
- [x] [generation kwargs](https://github.com/jmorganca/ollama/blob/main/docs/modelfile.md#valid-parameters-and-values) don't align with other standards (maybe we can forcefully set them to deployment config when running `ollama run model`?)

Things to note:
- we use the `api/chat` endpoint
- we can't do `n_generations` in one request so we make multiple

### CODE REVIEW

File: src/distilabel/llm/ollama.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+        try:
+            # Send the request
+            with request.urlopen(req) as response:
+                # Check if the request was successful (status code 200)
+                if response.getcode() == 200:
+                    # Parse and return the response JSON
+                    return json.loads(response.read().decode("utf-8"))
+                elif response.getcode() >= 500 and n < 5:
+                    # If the request failed, try again with backoff
```
Comment (plaguss) [inline src/distilabel/llm/ollama.py:102]: We already have backoff implemented with [tenacity](https://github.com/jd/tenacity), you can take a look at the inference endpoints from HF [here](https://github.com/argilla-io/distilabel/blob/main/src/distilabel/llm/huggingface/inference_endpoints.py#L193-L202) for reference.

**Review by plaguss — COMMENTED** [review]
Nice addition!! For when you tackle the docs, it would be nice to include an example of how to setup ollama so it is as contained as possible

**Comment by davidberenstein1957** [comment]
Difficulty w.r.t. `__init__` args was first defined here https://github.com/argilla-io/distilabel/issues/246.

**Review by plaguss — COMMENTED** [review]
Nice! Could you add some tests like the ones [here](https://github.com/argilla-io/distilabel/tree/main/tests/llm)? After that we should have a new cool integration 😃

**Pull request merged**